File: apps/api/src/main/java/com/st6/config/SeedData.java
package com.st6.config;

import com.st6.domain.CommitCategory;
import com.st6.domain.SupportingOutcome;
import com.st6.domain.WeeklyCommit;
import com.st6.domain.WeeklyPlan;
import com.st6.repository.SupportingOutcomeRepository;
import com.st6.repository.WeeklyPlanRepository;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;
import org.springframework.boot.CommandLineRunner;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class SeedData {
    @Bean
    CommandLineRunner seed(
            SupportingOutcomeRepository outcomeRepository, WeeklyPlanRepository planRepository) {
        return args -> {
            if (outcomeRepository.count() > 0) {
                return;
            }

            var outcome =
                    outcomeRepository.save(
                            SupportingOutcome.builder()
                                    .id(UUID.fromString("11111111-1111-1111-1111-111111111111"))
                                    .rallyCry("Win enterprise trust")
                                    .definingObjective("Make strategy execution auditable")
                                    .outcome(
                                            "Every weekly commitment maps to an approved Supporting Outcome")
                                    .owner("Strategy Ops")
                                    .build());

            var plan =
                    WeeklyPlan.builder()
                            .id(UUID.fromString("22222222-2222-2222-2222-222222222222"))
                            .weekStart(LocalDate.of(2026, 6, 22))
                            .ownerId("u-ava")
                            .ownerName("Ava Chen")
                            .managerId("u-morgan")
                            .build();
            plan.addCommit(
                    WeeklyCommit.builder()
                            .id(UUID.fromString("33333333-3333-3333-3333-333333333333"))
                            .supportingOutcome(outcome)
                            .title("Ship RCDO-linked commit form")
                            .description("Require Supporting Outcome mapping on every commit.")
                            .category(CommitCategory.QUEEN)
                            .priority(1)
                            .plannedHours(BigDecimal.valueOf(12))
                            .build());
            planRepository.save(plan);
        };
    }
}

File: apps/web/src/domain/seedData.ts
import type { ManagerDashboard, SupportingOutcome, WeeklyPlan } from "./types";

export const outcomes: SupportingOutcome[] = [
  {
    id: "so-101",
    rallyCry: "Win enterprise trust",
    definingObjective: "Make strategy execution auditable",
    outcome: "Every weekly commitment maps to an approved Supporting Outcome",
    owner: "Strategy Ops"
  },
  {
    id: "so-102",
    rallyCry: "Win enterprise trust",
    definingObjective: "Reduce manager review latency",
    outcome: "Managers complete weekly review within 24 business hours",
    owner: "People Ops"
  },
  {
    id: "so-201",
    rallyCry: "Scale the platform",
    definingObjective: "Harden PA micro-frontend extensibility",
    outcome: "New remotes load through Module Federation without shell coupling",
    owner: "Platform"
  },
  {
    id: "so-301",
    rallyCry: "Increase execution discipline",
    definingObjective: "Improve reconciliation accuracy",
    outcome: "Planned vs actual variance is visible for every team by Friday",
    owner: "Finance Ops"
  }
];

export const initialPlan: WeeklyPlan = {
  id: "plan-2026-06-22-ava",
  weekStart: "2026-06-22",
  ownerId: "u-ava",
  ownerName: "Ava Chen",
  managerId: "u-morgan",
  lifecycleState: "DRAFT",
  commits: [
    {
      id: "c-1",
      ownerId: "u-ava",
      ownerName: "Ava Chen",
      title: "Ship RCDO-linked commit form",
      description: "Replace free-text weekly plan entry with required Supporting Outcome mapping.",
      supportingOutcomeId: "so-101",
      category: "Queen",
      priority: 1,
      plannedHours: 12,
      actualHours: 0,
      status: "Planned"
    },
    {
      id: "c-2",
      ownerId: "u-ava",
      ownerName: "Ava Chen",
      title: "Expose remote entry contract",
      description: "Publish the weekly planning route as a Module Federation remote.",
      supportingOutcomeId: "so-201",
      category: "Rook",
      priority: 2,
      plannedHours: 7,
      actualHours: 0,
      status: "Planned"
    },
    {
      id: "c-3",
      ownerId: "u-ava",
      ownerName: "Ava Chen",
      title: "Prepare reconciliation variance view",
      description: "Show planned hours, actual hours, status, and carry-forward decisions.",
      supportingOutcomeId: "so-301",
      category: "Bishop",
      priority: 3,
      plannedHours: 6,
      actualHours: 0,
      status: "Planned"
    }
  ]
};

export const initialDashboard: ManagerDashboard = {
  weekStart: "2026-06-22",
  completionRate: 84,
  alignmentRate: 96,
  reviewTurnaroundHours: 18,
  teamMembers: [
    {
      id: "u-ava",
      name: "Ava Chen",
      role: "Senior Product Engineer",
      lifecycleState: "DRAFT",
      alignment: 100,
      plannedHours: 25,
      actualHours: 0,
      blockedCommits: 0
    },
    {
      id: "u-eli",
      name: "Eli Brooks",
      role: "Data Engineer",
      lifecycleState: "LOCKED",
      alignment: 92,
      plannedHours: 31,
      actualHours: 11,
      blockedCommits: 1
    },
    {
      id: "u-nora",
      name: "Nora Patel",
      role: "Engineering Manager",
      lifecycleState: "RECONCILING",
      alignment: 88,
      plannedHours: 28,
      actualHours: 23,
      blockedCommits: 0
    }
  ]
};
